c2rust-analyze/src/main.rs

```diff
@@ -440,76 +442,92 @@ fn run(tcx: TyCtxt) {
     // computed during this the process is kept around for use in later passes.
     let mut global_equiv = GlobalEquivSet::new(gacx.num_pointers());
     for &ldid in &all_fn_ldids {
+        // The function might already be marked as failed if one of its callees previously failed.
+        if gacx.fn_failed(ldid.to_def_id()) {
```

Comment:
would it be much extra work to run this in a fixed point loop instead? that way we can avoid the problem if `g` calling `f` where `f` is rewritten but `g` isn't and we get a compiler error. I can take that on if you feel like it's reasonable approach and can be in a separate PR regardless of whether or not that's the approach to take

Reply:
What's the change you're proposing here?  Currently failure is propagated "up" the callgraph (from callees to callers) in `AnalysisCtxt::mark_fn_failed`.  This part doesn't require a fixpoint, just a depth-first traversal of the callgraph.

The "easy" way to avoid having calls between rewritten and nonrewritten functions is to propagate failure in both directions, but (assuming there's no dead code) that means a failure anywhere will propagate to all parts of the program (first up the callgraph into `main`, then down the callgraph into everything else).

Reply:
I think you're right, and I hadn't considered the "easy" way either. Nevermind